### PR TITLE
Prevent dragging/dropping DOM elements from doing anything

### LINKED ISSUE #255: Suggestion: Option to prevent dragging and dropping elements on the page onto the dropzone
In the meantime, here is my solution for doing this:

```
onDragEnter={function( event ) {
    const dt = event.dataTransfer;
    if ( !(
        dt.types &&
        ( dt.types.indexOf ? dt.types.indexOf( "Files" ) !== -1 : dt.types.contains( "Files" ) )
    ) ) {
        this.setState( { isDragActive: false } );
    }
}}
```

```
onDrop={files => {
    if ( files.length === 0 ) {
        return;
    }

    // handle files dropped here
}}
```

### PR DESCRIPTION
**Pull request by buob**
Fixes #255

### CODE REVIEW

**Comment by buob** [comment]
Not sure why the PR isn't linked in the close link, but this fixes https://github.com/okonet/react-dropzone/issues/255

**Comment by buob** [comment]
Also, not sure why codecov thinks I'm losing coverage, I'm adding quite a few more LOC of tests than actual code. 😞

**Comment by buob** [comment]
@okonet Is there any update on this?  Is there something I need to change to make this a merge candidate?  I assumed at the time of submitting codecov was acting up, but if it's an actual problem I have, let me know 👍.

**Comment by okonet** [comment]
I've received a second PR that should fix this issue: https://github.com/okonet/react-dropzone/pull/340 and from looking at the code I prefer it since it introduces much less new code. What are your thoughts? Does the second one missing some edge-cases that aren't covered in tests?

File: src/index.js [hunk 1]
```diff
@@ -101,6 +109,11 @@ class Dropzone extends React.Component {
     // Stop default browser behavior
     e.preventDefault();
 
+    // If there are no files (i.e. if you drag a DOM element or image from within the browser)
+    if (fileList.length === 0) {
+      return;
```
Comment (okonet) [inline src/index.js]: I feel like we still want to reset `this.enterCounter` etc in this case.

File: src/index.js [hunk 2]
```diff
@@ -44,6 +44,14 @@ class Dropzone extends React.Component {
   onDragEnter(e) {
     e.preventDefault();
 
+    const dt = e.dataTransfer;
+
+    // If the object isn't a file (dom element, link, etc.)
+    if (!(dt.types && (dt.types.indexOf ? dt.types.indexOf('Files') !== -1 : dt.types.contains('Files')))) {
```
Comment (okonet) [inline src/index.js]: `.contains` isn't implemented in most of the browsers yet AFAIK. I'd rather had this extracted to a util function with separate tests.

**Comment by buob** [comment]
Hi @okonet!  My impetus for opening this PR was actually more that in Safari if you drag a DOM element (i.e. an image on the page) over the target, it triggers the drag active class.  The other PR doesn't address that.

I'm happy to investigate the other approach for the dropping portion though, it certainly is cleaner. 👍

**Comment by okonet** [comment]
Yes, can you please add a description of which exact problems it is solving?

File: src/index.spec.js [hunk 3]
```diff
@@ -636,14 +643,37 @@ describe('Dropzone', () => {
...
+        const dropzone = mount(
+          <Dropzone onDragEnter={onDragEnter} />
+        );
+
+        dropzone.simulate('dragEnter', { dataTransfer: { files: [], types: ['text/uri-list'] } });
+        expect(dropzone.state().isDragActive).toEqual(false);
+        expect(onDragEnter.called).not.toBeTruthy();
+      });
+
+      it.only('does not drop the items', () => {
```
Comment (masylum) [inline src/index.spec.js]: ☝️

**Comment by seberik** [comment]
@okonet Why did this become closed? I would be happy to continue working on a fix for this issue as we solved it with a onDragStart workaround. The problem is that we have a html5-drag-and-drop component inside of the dropzone component to allow images to be uploaded and sorted. When users sorts the uploaded images react-dropzone calls onDropRejected with a dataTransferItem of type "application/json".

**Comment by rkmarks** [comment]
@seberik Correct, the issue was closed due to inactivity as mentioned by the stale bot. You can continue to work on your fix and make a PR whenever you can :)
